pkg/server/server.go

```diff
@@ -177,8 +181,37 @@ func (p *Plugin) Attest(stream nodeattestorv1.NodeAttestor_AttestServer) error {
 			}
 		}
 
+		// x509 library can not parse the SANs from the EKcert so we use x509ext instead
+		for _, ext := range ek.Certificate.Extensions {
+			if ext.Id.Equal(oid.SubjectAltName) {
+				_, err := x509ext.ParseSubjectAltName(ext)
+				if err != nil {
+					return status.Errorf(codes.InvalidArgument, "tpm: could not parse SANs of EKCertt : %v", err)
+				}
+			}
+		}
+
+		oidTcgKpEKCertificate := asn1.ObjectIdentifier{2, 23, 133, 8, 1}
+		for _, id := range ek.Certificate.UnknownExtKeyUsage {
+			if id.Equal(oidTcgKpEKCertificate) {
+			}
+		}
```

Comment:
Dead code?

Reply:
oh no I think we should throw an error here if it's *not* this key usage. Let me add that.